sycl/source/detail/memory_manager.cpp

```diff
@@ -225,19 +240,31 @@ void copyD2H(SYCLMemObjT *SYCLMemObj, cl_mem SrcMem, QueueImplPtr SrcQueue,
   if (1 == DimDst && 1 == DimSrc) {
     CHECK_OCL_CODE(clEnqueueReadBuffer(
         CLQueue, SrcMem,
-        /*blocking_read=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
-        DstMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
+	/*blocking_read=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
+	DstMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
   } else {
-    size_t BufferRowPitch = (1 == DimSrc) ? 0 : SrcSize[0];
-    size_t BufferSlicePitch = (3 == DimSrc) ? SrcSize[0] * SrcSize[1] : 0;
 
-    size_t HostRowPitch = (1 == DimDst) ? 0 : DstSize[0];
-    size_t HostSlicePitch = (3 == DimDst) ? DstSize[0] * DstSize[1] : 0;
-    CHECK_OCL_CODE(clEnqueueReadBufferRect(
-        CLQueue, SrcMem,
-        /*blocking_read=*/CL_FALSE, &SrcOffset[0], &DstOffset[0],
-        &SrcAccessRange[0], BufferRowPitch, BufferSlicePitch, HostRowPitch,
-        HostSlicePitch, DstMem, DepEvents.size(), &DepEvents[0], &OutEvent));
+    if( SrcSize[0]   == SrcAccessRange[0] &&  SrcSize[1]   == SrcAccessRange[1]  &&  SrcSize[2]   == SrcAccessRange[2] &&
+	SrcSize[0]   == DstSize[0]        &&  SrcSize[1]   == DstSize[1]         &&  SrcSize[2]   == DstSize[2]        &&
+	SrcSize[0]   == DstAccessRange[0] &&  SrcSize[1]   == DstAccessRange[1]  &&  SrcSize[2]   == DstAccessRange[2] &&
+	SrcOffset[0] == DstOffset[0]      &&  SrcOffset[1] == DstOffset[1]       &&  SrcOffset[2] == DstOffset[2]      &&
+	SrcOffset[0] == 0                 &&  SrcOffset[1] == 0                  &&  SrcOffset[2] == 0 ) {
```

Comment:
This looks like déjà vu.
Put this in a `constexpr` function such as `is_compact_transfer()` or `is_trivial_transfer()` or simple or...?